src/navigation/reroute.js

```diff
@@ -32,10 +32,10 @@ export function reroute(pendingPromises = [], eventArguments) {
     });
   }
 
-  appChangeUnderway = true;
   let wasNoOp = true;
 
   if (isStarted()) {
+    appChangeUnderway = true;
```

Comment:
`appChangeUnderway` is a concurrency lock. When its true, everything waits for the current promise to finish before doing the next thing.

We only want to the lock to occur after `singleSpa.start()` has occurred since `load()` is the only thing that can happen before start() is called and it's (now) okay to call `load()` multiple times on the same application.

Reply:
What consequences does calling load multiple times have. I see that the tests (both old and new) pass and I’d like further clarification on the consequences of calling load multiple times. 

If I’m understanding correctly it shouldn’t matter because we store the loading promise so it’s the same promise but occasionally we delete the promise, what happens if load is called multiple times and the promise has been deleted?

Reply:
> If I’m understanding correctly it shouldn’t matter because we store the loading promise so it’s the same promise but occasionally we delete the promise,

Yes, exactly

> what happens if load is called multiple times and the promise has been deleted?

I don't think that this should ever happen, although it's a good thing to think about. `toLoadPromise()` is only ever called when `getAppsToLoad()` returns the app name. And that is dependent on the application status. Since the `app.loadPromise` only ever exists when the application is in LOADING_SOURCE_CODE status, and since getAppsToLoad() doesn't return applications that are in LOADING_SOURCE_CODE status, I don't think it's possible for `toLoadPromise()` to ever be called after the promise is deleted (unless you call `reloadApplication()`.

^ That paragraph is dense enough to confuse even me, the author of it :). I am not trying to hand wave here - your question is a good one. If there's a problem with this change it would be a lot better to catch it now than after we publish a new version. Can you think of a situation where `toLoadPromise` would be called after the `app.loadPromise` is deleted and after the application has already loaded?

Reply:
Only reload application, that's not a common API but I do want to verify that this works with that.

Reply:
This test verifies that - https://github.com/single-spa/single-spa/blob/e8dfb1319267e6b7cce6ade43120557b0f19b045/spec/apps/invalid-load-function/invalid-load-function.spec.js#L99

It was failing when I first made the changes until I verified that appChangeUnderway was only resetting to false when it should

Reply:
Actually this test might be a better one to look at - https://github.com/single-spa/single-spa/blob/19a8ecd6f328323ab64978125d8137a0d463a71e/spec/apps/happy-unload/happy-unload.spec.js